server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -466,7 +462,7 @@ private void sendClusterState(
             transportService.sendChildRequest(
```

Comment:
I'd also rather we specified the exact `Transport.Connection` here. This will be a somewhat more involved change because we'll need to do our own tracking of the relationship between `DiscoveryNode` and `Transport.Connection` throughout the whole `PublicationContext`.

Reply:
Turns out it wasn't that complicated to do